src/os/filestore/FileStore.cc

```diff
@@ -2530,9 +2530,10 @@ void FileStore::_do_transaction(
       break;
     case Transaction::OP_TOUCH:
       {
-        coll_t cid = i.get_cid(op->cid);
+        const coll_t &_cid = i.get_cid(op->cid);
         const ghobject_t &oid = i.get_oid(op->oid);
-	_kludge_temp_object_collection(cid, oid);
+        const coll_t &cid = !_need_temp_object_collection(_cid, oid) ?
+          _cid : _cid.get_temp();
```

Comment:
yes, it's valid. see http://stackoverflow.com/questions/2784262/does-a-const-reference-prolong-the-life-of-a-temporary